include/sparrow/config.hpp

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#if defined(__apple_build_version__)
+#define COMPILING_WITH_APPLE_CLANG
+#endif
+
+consteval bool is_apple_compiler()
```

Comment:
Would it makes sense to have it `constinit`?

Reply:
`constinit` is only for objects and is forcing the initial value to be evaluated at compile-time. Here `consteval` means this fonction can only be called at compile-time (therefore, in "constant expressions" that is code which runs at compile-time). If it was an object instead of a function then  `constexpr` (defined at compile time, can be accessed at runtime) would be best, as `consteval` (exists only at compile-time) is not (yet) allowed on anything other than functions.